src/mapgen/mg_decoration.cpp

```diff
@@ -127,6 +122,19 @@ bool Decoration::canPlaceDecoration(MMVManip *vm, v3s16 p)
 			nneighs++;
 	}
 
+	if (check_offset != 0) {
+		const v3s16 dir_offset = v3s16( 0, 1 + check_offset,  0);
```

Comment:
```suggestion
		const v3s16 dir_offset = v3s16(0, check_offset, 0);
```
You're adding the offset to the Y=1 values above, so to cover Y=0 with check_offset=-1, this needs correcting.

Reply:
done